x-pack/test/api_integration/apis/telemetry/telemetry_local.js

```diff
@@ -76,10 +76,10 @@ export default function ({ getService }) {
       expect(stats.stack_stats.kibana.plugins.apm.services_per_agent).to.be.an('object');
       expect(stats.stack_stats.kibana.plugins.infraops.last_24_hours).to.be.an('object');
       expect(stats.stack_stats.kibana.plugins.kql.defaultQueryLanguage).to.be.a('string');
-      expect(stats.stack_stats.kibana.plugins.maps.timeCaptured).to.be.a('string');
-      expect(stats.stack_stats.kibana.plugins.maps.attributes).to.be(undefined);
-      expect(stats.stack_stats.kibana.plugins.maps.id).to.be(undefined);
-      expect(stats.stack_stats.kibana.plugins.maps.type).to.be(undefined);
+      expect(stats.stack_stats.kibana.plugins['maps-telemetry'].timeCaptured).to.be.a('string');
+      expect(stats.stack_stats.kibana.plugins['maps-telemetry'].attributes).to.be(undefined);
+      expect(stats.stack_stats.kibana.plugins['maps-telemetry'].id).to.be(undefined);
+      expect(stats.stack_stats.kibana.plugins['maps-telemetry'].type).to.be(undefined);
```

Comment:
I sincerely don't think the savedObjects type should affect the name of the reported telemetry.
How we obtain the telemetry to be returned in the `fetch` method in the `usageCollector` (in this case, reading the `maps-telemetry` SO) should not affect the name of the reported telemetry `maps`: 
Simply change the line [`type: TELEMETRY_TYPE,`](https://github.com/aaronjcaldwell/kibana/blob/rollback-telemetry-saved-object-change-and-remove-dynamic-fields/x-pack/plugins/maps/server/maps_telemetry/collectors/register.ts#L22) to `type: 'maps',`.

Reply:
I've changed it to have the saved object as `maps_telemetry` and the `usageCollector` type as `maps` as you've described.

> Another proof of it, it's the getMapSavedObjects method is using the constant MAP_SAVED_OBJECT_TYPE, which happens to be maps

I don't quite follow you on this one. `MAP_SAVED_OBJECT_TYPE` is actually `map` singular, so not looking at the same thing. This is likely why we named `maps_telemetry` as it was named in the first place, to not have `maps` and `map` SOs.

Reply:
I think you are right `MAP_SAVED_OBJECT_TYPE` is not related at all. Maybe we are not storing the `maps-telemetry` SavedObjects at all anymore?

I'll clone this branch after lunch and I'll give it a go to fully understand what SavedObjects are for what :)

Reply:
As a follow-up question, how is telemetry keeping track of the maps saved object type `maps-telemetry`? On our end this feels a little disconnected. We're registering a collector with type `maps` that collects data conforming to the shape of the saved object `maps-telemetry` that we define in our mappings. It's fine to have the names not be 1:1, but the connection between the two isn't clear to me

Reply:
Our messages crossed paths but I think we're more or less saying the same thing. Maybe this just needs to be removed then

Reply:
I thought previously this might've been used for some old task manager logic, but [looking back at that](https://github.com/elastic/kibana/blob/7.4/x-pack/legacy/plugins/maps/server/maps_telemetry/telemetry_task.js) it shouldn't have needed a saved object there either since really we were just grabbing telemetry 1x/24 hours and writing it to task manager state, no SO required so far as I can tell. When we first implemented, it was mostly just copy and paste from existing sources and then updates regarding the shape of our data. It looks like I may have just stored an extra saved object along the way that should be removed, my mistake. :nauseated_face: 

If it's not being used on your end, which it sounds like it's not, I say we just remove it

Reply:
Yeah! I think it evolved from time to time: Initially, the task manager logic was generating the saved object and the `fetch` method was simply returning it.
Then the task manager logic was removed but we kept the saved object in place.
But, to make it clear: the `usageCollection`/`telemetry` plugin does not rely on any savedObjects created by the registered `usageCollector`s. It simply calls the `fetch` method and appends the returned payload into the _big_ telemetry payload under `stack_stats.kibana.plugins[type]`.
In pseudocode:

```typescript
for (const collector of registeredUsageCollectors) {
  fullTelemetryPayload.stack_stats.kibana.plugins[collector.type] = await collector.fetch(callCluster);
}

await http.post(TELEMETRY_URL, { body: fullTelemetryPayload });
```

Reply:
From the telemetry POV, I'd say it's safe to remove that mappings definition :)

Reply:
Removed!